src/rdkafka_queue.c

```diff
@@ -537,6 +565,7 @@ int rd_kafka_q_serve_rkmessages (rd_kafka_q_t *rkq, int timeout_ms,
         rd_kafka_t *rk = rkq->rkq_rk;
         rd_kafka_q_t *fwdq;
         struct timespec timeout_tspec;
+        rd_kafka_op_t *rkos[rkmessages_size];
```

Comment:
Don't think we need to maintain a separate array for the rkos. The rkmessage is part of the rko allocation, and rkmessage->_private points back to the rko, so we can just use that in the purge function.